metrics-exporter-log/src/lib.rs

```diff
@@ -66,23 +78,16 @@ where
         Interval::new_interval(interval)
             .map_err(|_| ())
             .for_each(move |_| {
-                let recorder = recorder.clone();
-                run_once(&controller, recorder, level);
-                Ok(())
-            })
-    }
-}
+                let mut recorder = recorder.clone();
 
-fn run_once<R>(controller: &Controller, mut recorder: R, level: Level)
-where
-    R: MetricsRecorder + Into<String>
-{
-    match controller.get_snapshot() {
-        Ok(snapshot) => {
-            snapshot.record(&mut recorder);
-            let output = recorder.into();
-            log!(level, "{}", output);
-        },
-        Err(e) => log!(Level::Error, "failed to capture snapshot: {}", e),
+                controller.get_snapshot_async()
+                    .and_then(move |snapshot| {
+                        snapshot.record(&mut recorder);
+                        let output = recorder.into();
+                        log!(level, "{}", output);
+                        Ok(())
+                    })
+                    .map_err(|e| log!(Level::Error, "failed to get snapshot: {}", e))
```

Comment:
`error!`?

Reply:
Yeah, that's fair.  I was just following the `log!` usage above it.  I'll change it for succinctness, though.